repos/system_upgrade/el8toel9/actors/dotnet/tests/test_dotnet.py

```diff
@@ -0,0 +1,47 @@
+import pytest
+
+from leapp import reporting
+from leapp.libraries.common.testutils import create_report_mocked, CurrentActorMocked
+from leapp.libraries.stdlib import api
+from leapp.models import InstalledRPM, RPM
+
+def _generate_rpm_with_name(name):
+    return RPM(name=name,
+               version='0.1',
+               release='1.sm01',
+               epoch='1',
+               pgpsig='RSA/SHA256, Mon 01 Jan 1970 00:00:00 AM -03, Key ID 199e2f91fd431d51',
+               packager='Red Hat, Inc. <http://bugzilla.redhat.com/bugzilla>',
+               arch='noarch')
+
+@pytest.mark.parametrize('unsupported_versions', [
+    ([]),          # No unsupported versions
+    ([2.1]),       # Single unsupported version
+    ([3.0]),       # Other unsupported version
+    ([2.1, 3.0]),  # Multiple unsupported versions
+])
+def test_actor_execution(monkeypatch, unsupported_versions):
+    """
+    Install one or more dotnet-runtime packages for unsupported versions
+    and verify a report is generated.
+    """
+
+    # Couple of random packages
+    rpms = [_generate_rpm_with_name('sed'),
+            _generate_rpm_with_name('htop')]
+
+    # dotnet-runtime-{version} packages
+    for version in installed_versions:
+        rpms += [_generate_rpm_with_name(f'dotnet-runtime-{version}')]
+
+    curr_actor_mocked = CurrentActorMocked(msgs=[InstalledRPM(items=rpms)])
+    monkeypatch.setattr(api, 'current_actor', curr_actor_mocked)
+    monkeypatch.setattr(reporting, "create_report", create_report_mocked())
```

Comment:
Sorry for nitpicking, could you please use one type of quotes for strings in the module? Either " or '?

Reply:
Actually we use single quotes for everything but doc strings, there might be some exceptions for some existing code that is used as a library, but we do usually insist on one style of quotes in the repository.